### PR TITLE
Add Shark Benchmark

### PR DESCRIPTION
**Pull request by raikonenfnu**
-Introduce SharkBenchmark that bench models on regular torch, shark-py, and shark-c.
-Integrate iree-benchmark-module into Shark.
-Refactor SharkRunner to have low level API control.

### CODE REVIEW

**Comment by powderluv** [comment]
Push and Iterate.

**Comment by pashu123** [comment]
One question: Why can't benchmark be a function under SharkInference? Or just a SharkBenchMarkRunner just like SharkRunner that can be called inside SharkInference - so instead of forward we can call the benchmark function.

**Comment by raikonenfnu** [comment]
> One question: Why can't benchmark be a function under SharkInference? Or just a SharkBenchMarkRunner just like SharkRunner that can be called inside SharkInference - so instead of forward we can call the benchmark function.

Integrated it into ShakRunner :)

**Review by powderluv — APPROVED** [review]

File: shark/iree_utils.py [hunk 1]
```diff
@@ -152,6 +159,15 @@ def get_iree_compiled_module(module,
 
     return get_iree_module(module, device, input_type, args, func_name)
 
+def export_iree_module_to_vmfb(module, device: str, directory: str):
```
Comment (powderluv) [inline shark/iree_utils.py:162]: Nit:Shark module? In our code we can call IREE SHARK

**Review by powderluv — APPROVED** [review]

File: shark/iree_utils.py (hunk 1 above)
Comment (raikonenfnu) [reply shark/iree_utils.py:162]: There's actually a bunch of these in iree_utils, I'll push this up in a separate patch! :)

**Pull request merged**